airflow/executors/mesos_executor.py

```diff
@@ -0,0 +1,469 @@
+
+        if task_state == "TASK_RUNNING":
+            self.result_queue.put((key, State.RUNNING))
+            return
+
+        if task_state == "TASK_FINISHED":
+            self.result_queue.put((key, State.SUCCESS))
+            return
+
+        if task_state == "TASK_LOST" or \
+           task_state == "TASK_KILLED" or \
+           task_state == "TASK_FAILED":
+            self.result_queue.put((key, State.FAILED))
+            return
+
+
+class MesosExecutor(BaseExecutor):
+    """
+    MesosExecutor allows distributing the execution of task
+    instances to multiple mesos workers.
+
+    Apache Mesos is a distributed systems kernel which abstracts
+    CPU, memory, storage, and other compute resources away from
+    machines (physical or virtual), enabling fault-tolerant and
+    elastic distributed systems to easily be built and run effectively.
+    See http://mesos.apache.org/
+    """
+
+    class MesosFramework(threading.Thread):
+        """MesosFramework class to start the threading"""
+
+        def __init__(self, client):
+            threading.Thread.__init__(self)
+            self.client = client
+            self.stop = False
+
+        def run(self):
+            try:
+                self.client.register()
+            except KeyboardInterrupt:
+                self.log.info("Stop requested by user, stopping framework....")
+
+    def __init__(self):
+        super().__init__()
+        self.commands_to_run = []
+        self.task_queue = Queue()
+        self.result_queue = Queue()
+        self.driver = None
+        self.client = None
+        self.th = None  # pylint: disable=invalid-name
+
+    @provide_session
+    def start(self, session=None):
+        """Setup and start routine to connect with the mesos master"""
+        if not conf.get('mesos', 'MASTER'):
+            self.log.error("Expecting mesos master URL for mesos executor")
+            raise AirflowException("mesos.master not provided for mesos executor")
+
+        master = conf.get('mesos', 'MASTER')
+
+        framework_name = get_framework_name()
+        framework_id = None
+
+        task_cpu = conf.get('mesos', 'TASK_CPU', fallback=1)
+        task_memory = conf.get('mesos', 'TASK_MEMORY', fallback=256)
+
+        if conf.getboolean('mesos', 'CHECKPOINT'):
+            framework_checkpoint = True
+
+            if conf.get('mesos', 'FAILOVER_TIMEOUT'):
+                # Import here to work around a circular import error
+                from airflow.models import Connection
+
+                # Query the database to get the ID of the Mesos Framework, if available.
+                conn_id = FRAMEWORK_CONNID_PREFIX + framework_name
+                connection = session.query(Connection).filter_by(conn_id=conn_id).first()
+                if connection is not None:
+                    # Set the Framework ID to let the scheduler reconnect
+                    # with running tasks.
+                    framework_id = connection.extra
```

Comment:
```suggestion
                    framework_id = connection.extra
```
`connection.extra` is a string, is this expected?

Reply:
yes it is. why not? :-) did I misunderstood something?

Reply:
If this is planned then ok. However, usually `extra` field is used to keep for example a dictionary (as JSON) so you can use `conn.extra_dejson` and then access the values by keys. This provides more structured way of keeping the information

Reply:
thats a good idea. :+1: